src/main/java/com/faforever/client/player/PlayerServiceImpl.java

```diff
@@ -40,11 +62,52 @@ void init() {
     lobbyServerAccessor.setOnPlayerInfoMessageListener(this);
     lobbyServerAccessor.setOnFoeListListener(this);
     lobbyServerAccessor.setOnFriendListListener(this);
+    gameService.addOnGameInfoBeanListener(change -> {
+      while (change.next()) {
+        for (GameInfoBean gameInfoBean : change.getRemoved()) {
+          gameInfoBean.setStatus(GameState.CLOSED);
+          gameInfoBean.getTeams().forEach((team, players) -> updatePlayerInfoBean(players, gameInfoBean));
+        }
+        for (GameInfoBean gameInfoBean : change.getAddedSubList()) {
+          gameInfoBean.getTeams().forEach((team, players) -> updatePlayerInfoBean(players, gameInfoBean));
+          gameInfoBean.statusProperty().addListener(change2 -> {
+            gameInfoBean.getTeams().forEach((team, updatedPlayer) -> updatePlayerInfoBean(updatedPlayer, gameInfoBean));
+          });
+        }
+      }
+    });
+  }
+
+  //FIXME ugly fix until host can be resolved from gamestate
+  private void updatePlayerInfoBean(List<String> players, GameInfoBean gameInfoBean) {
+    for (String player : players) {
+      PlayerInfoBean playerInfoBean = getPlayerForUsername(player);
+      if(playerInfoBean == null) {
+        continue;
+      }
+      updatePlayerGameStatus(playerInfoBean, GameStatus.getFromGameState(gameInfoBean.getStatus()));
+      playerInfoBean.setGameUID(gameInfoBean.getUid());
+    }
+    if (GameStatus.getFromGameState(gameInfoBean.getStatus()) == GameStatus.LOBBY) {
+      PlayerInfoBean host = getPlayerForUsername(gameInfoBean.getHost());
+      updatePlayerGameStatus(host, GameStatus.HOST);
+    }
+  }
+
+  @Override
+  public void updatePlayerGameStatus(PlayerInfoBean playerInfoBean, GameStatus gameStatus) {
+    if (playerInfoBean != null && playerInfoBean.getGameStatus() != gameStatus) {
+      //FIXME until api, host is set twice or ugly code, I chose host set twice
+      playerInfoBean.setGameStatus(gameStatus);
+    }
   }
 
   @Override
   public PlayerInfoBean getPlayerForUsername(String username) {
-    return players.get(username);
+    PlayerInfoBean playerInfoBean = players.get(username);
+    if(playerInfoBean == null)
+      logger.warn("Error returning playerInfoBean of {}", username);
```

Comment:
"Error returning.." sounds like "something went wrong" in the sense of an exception or so, I'd say "Unknown user: {}"